src/utils/urlParams/index.ts

```diff
@@ -96,14 +96,33 @@ interface NavigateOptions {
 
 export const glifParams = ['network']
 
+export const isNetworkInPath = (asPath: string) => {
+  // our apps can support both glif.io/<network>/address/<address> AND glif.io/address/<address>/?network=<network>
+  // when we switch networks, we want to make sure we stick with the same routing protocol
+  const supposedNetwork = asPath.split('/')[1]
+  // if the network name was encoded in path we have to rebuild the URL without query params
+  return Object.values(Network).some(
+    net => net === supposedNetwork || net === `${supposedNetwork}net`
+  )
+}
+
 export function navigate(
   router: NextRouter,
   { pageUrl, params, retainParams }: NavigateOptions
 ): void {
   const query = router?.query
   let updatedUrl = pageUrl
 
-  if (query) {
+  if (query && isNetworkInPath(router.asPath)) {
+    updatedUrl += `/${query.network}`
```

Comment:
I'm a bit confused here. Isn't the network always the first part of the path? E.g. `/calibrationnet/message` and not `/message/calibrationnet`. Also, if `updatedUrl` ends with a `/`, there could be a double slash in the url, I think we should prevent this by checking `updatedUrl.endsWith('/')`